Client/Frontend/Home/FirefoxHomeViewController.swift

```diff
@@ -581,11 +581,37 @@ extension FirefoxHomeViewController: DataObserverDelegate {
             
             let maxItems = Int(numRows) * self.topSitesManager.numberOfHorizontalItems()
             
-            self.topSitesManager.content = Array(result.prefix(maxItems))
+            var sites = Array(result.prefix(maxItems))
             
+            // Check if all result items are pinned site
+            var pinnedSites = 0
+            result.forEach {
+                if let _ = $0 as? PinnedSite {
+                    pinnedSites += 1
+                }
+            }
+            // Special case: Adding Google topsite
+            let googleTopSite = GoogleTopSiteHelper(prefs: self.profile.prefs)
+            let isIpad: Bool = UIDevice.current.userInterfaceIdiom == .pad
+            let isPortrait: Bool = UIApplication.shared.statusBarOrientation.isPortrait
+            let maxNumberOfTopSites = isIpad ? (isPortrait ? 12 : 16) : 8
```

Comment:
This breaks the setting for a custom number of top sites rows, `maxItems` should be used instead.

Reply:
Thanks for catching this @q2r5, going to push a chance to use `maxItems` that would avoid doing this logic entirely 🤞🏼